katcp/resource_client.py

```diff
@@ -735,6 +735,208 @@ def issue_request(self, *args, **kwargs):
         """
         return self._client.wrapped_request(self.name, *args, **kwargs)
 
```

Comment:
I'm not happy with all this being moved from resource.py to client_resource.py.  The design idea is:

resource.py specifies the abstract interface of what a KATCP resource is supposed to look like
client_resource.py specifies an implementation that uses direct KATCP connections.

The nice thing about having abstract classes is also that certain functionalities can be kept abstract in that they only depend on the implementors implementing a certain subset of the whole abstract API, see for instance the ABCs in python's builting collections module.